### PR TITLE
Use global ZDOTDIR environment variable

### PR DESCRIPTION
**Pull request by pdamianik**
Fallback to the users home directory if the variable is not set

Related issue: #159785

### CODE REVIEW

**Review by meganrogge — APPROVED** [review]

File: src/vs/workbench/contrib/terminal/browser/media/shellIntegration-env.zsh [hunk 1]
```diff
@@ -2,13 +2,12 @@
...
+	VSCODE_ZDOTDIR=$ZDOTDIR
+	ZDOTDIR=$USER_ZDOTDIR
+
+	. $USER_ZDOTDIR/.zshenv
+
 	USER_ZDOTDIR=$ZDOTDIR
 	ZDOTDIR=$VSCODE_ZDOTDIR
-else
-	USER_ZDOTDIR=$HOME
 fi
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/media/shellIntegration-env.zsh:13]: Am I understanding this right in that it fixes the case where `$ZDOTDIR` (eg. `/home/user/myzsh`) is set in your profile/login scripts and then have a `$ZDOTDIR/.zshenv` in that, not in `~/.zshenv`?

A comment explaining what's going on here would be good.

File: src/vs/platform/terminal/node/terminalEnvironment.ts [hunk 2]
```diff
@@ -186,6 +187,8 @@ export function getShellIntegrationInjection(
 			// Move .zshrc into $ZDOTDIR as the way to activate the script
 			const zdotdir = path.join(os.tmpdir(), `${os.userInfo().username}-vscode-zsh`);
 			envMixin['ZDOTDIR'] = zdotdir;
+			const userZdotdir = env?.ZDOTDIR ?? os.homedir();
+			envMixin['USER_ZDOTDIR'] = userZdotdir;
```
Comment (Tyriar) [inline src/vs/platform/terminal/node/terminalEnvironment.ts:191]: A comment explaining why this is needed would be good

File: src/vs/workbench/contrib/terminal/browser/media/shellIntegration-env.zsh (hunk 1 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/media/shellIntegration-env.zsh:13]: @Tyriar I believe actually what this solves for is when `~/.zshenv` is in a `$ZDOTDIR` which is not equal to `~` as we previously assumed it to be

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/media/shellIntegration-env.zsh:13]: which is why the user's `ZDOTDIR` must be captured before these scripts get run

File: src/vs/platform/terminal/node/terminalEnvironment.ts (hunk 2 above)
Comment (meganrogge) [reply src/vs/platform/terminal/node/terminalEnvironment.ts:191]: same explanation as above

File: src/vs/workbench/contrib/terminal/browser/media/shellIntegration-env.zsh (hunk 1 above)
Comment (Tyriar) [reply src/vs/workbench/contrib/terminal/browser/media/shellIntegration-env.zsh:13]: So `ZDOTDIR` gets set in `~/.profile`, or somewhere else I'm not thinking of?

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/media/shellIntegration-env.zsh:13]: set outside of zsh rc files
https://github.com/microsoft/vscode/issues/159785#issuecomment-1241088547

Comment (pdamianik) [reply src/vs/workbench/contrib/terminal/browser/media/shellIntegration-env.zsh:13]: > @Tyriar I believe actually what this solves for is when `~/.zshenv` is in a `$ZDOTDIR` which is not equal to `~` as we previously assumed it to be

Exactly

**Review by Tyriar — APPROVED** [review]
Lgtm

**Pull request merged**